docs/proposals/operator-custom-crd-download-url/README.md

```diff
@@ -0,0 +1,79 @@
+is based on the intended version of that instance, not the release version of the operator.
+
+## Proposal
+
+Currently, when provisioning a new Karmada instance, as a prerequisite, the Karmada operator will ensure that required CRDs are downloaded. By default, the CRDs tarball is
+downloaded from a well-known location on GitHub based on the release version of the operator. As aforementioned, this is based on the assumption that the version of the Karmada instance 
+is identical to that of the operator's release version, which may not always be true. When the version of the Karmada control plane components is not specified, the default behavior is to assume
+those components should have the same version as the operator's release version. In that scenario, the assumption is justified and holds true. However, when creating a new Karmada instance, it's possible to explicitly
+specify the version of the control plane components. In that scenario, this assumption breaks, and we encounter the risk of the operator installing the wrong CRDs for that Karmada instance.
+
+The default behavior is sensible and will be retained. However, it should not be the behavior when the intent is to create a new Karmada instance with control plane components of a different version than that of the operator's release version. 
+To support that scenario, we require the ability to explicitly set, as part of the Karmada resource, the URL that should be used to download the CRDs tarball for that Karmada instance.
+
+In addition to making sure required CRDs are downloaded, the Karmada operator also employs a caching strategy to avoid re-downloading CRDs it has already downloaded. It's a cache on the local file system.
+Downloaded CRDs are stored at a well-known location on the local file system based the release version of the Karmada operator. When provisioning a new Karmada instance, before attempting to download the required CRDs, the operator will fist check if
+they already exist in the cache. To support custom CRD download URLs, we require this caching strategy be extended to be more generic.
+
+## Design Details
+
+To support downloading CRDs from a custom URL, we need to add a new field called `CRDRemoteURL` to the Karmada resource spec:
+```go
+type KarmadaSpec struct {
+	HostCluster *HostCluster `json:"hostCluster,omitempty"`
+
+	PrivateRegistry *ImageRegistry `json:"privateRegistry,omitempty"`
+	
+	Components *KarmadaComponents `json:"components,omitempty"`
+	
+	FeatureGates map[string]bool `json:"featureGates,omitempty"`
+
+	// Custom URL that should be used to download required CRDs for this Karmada instance.
+	// Omit if the intent is to provision a new Karmada instance with control plane components of the same version as the Karmada operator's release version.
+	// Otherwise, it should be explicitly set to ensure the right version of the CRDs is installed for this instance.
+	// +optional
+	CRDRemoteURL string `json:"crdRemoteURL,omitempty"`
```

Comment:
1. We can add more comments describing where the tarball will be downloaded by default.
2. And, shall we consider adding the default value to this during the operator starts to reconcile the object?

Reply:
I wonder if we should leave room for future extensions.
For now, this field just requires a URL for the operator to download the tarball through `http`, shall we reserve the room for other protocols, like FTP,SFTP, etc.

Reply:
In case of the tarball exists in the cache, shall we always take it from the cache or download it from the URL?
Shall we introduce the download policy or something, like `Always` means always download from URL and `IfNotPresent` means will check the cache first, and then download it only when the cache missing.